From e00229cbc177ec3bf967ba40422a5bbf14837f79 Mon Sep 17 00:00:00 2001
From: Akira Ajisaka <aajisaka@apache.org>
Date: Fri, 8 Jul 2016 14:39:11 -0700
Subject: [PATCH 1693/2748] HDFS-10336. TestBalancer failing intermittently
 because of not reseting UserGroupInformation
 completely. Contributed by Yiqun Lin.

(cherry picked from commit 932aed64d77edcc8483a95c1ce31a4c9ae679446)

Change-Id: I50a11c742e6398b6567f6468dd2bb6efdf6f8d45
---
 .../hadoop/hdfs/server/balancer/TestBalancer.java  |    1 +
 1 file changed, 1 insertion(+)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/balancer/TestBalancer.java b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/balancer/TestBalancer.java
index b97e7cc..613cf06 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/balancer/TestBalancer.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/balancer/TestBalancer.java
@@ -1609,6 +1609,7 @@ public Void run() throws Exception {
       });
     } finally {
       // Reset UGI so that other tests are not affected.
+      UserGroupInformation.reset();
       UserGroupInformation.setConfiguration(new Configuration());
     }
   }
-- 
1.7.9.5

